src/NServiceBus.Core/Sagas/SagaPersistenceBehavior.cs

```diff
@@ -46,30 +46,62 @@ public SagaPersistenceBehavior(ISagaPersister persister, ICancelDeferredMessages
                 return;
             }
 
-            var sagaMetadata = sagaMetadataCollection.Find(context.MessageHandler.Instance.GetType());
-            var sagaInstanceState = new ActiveSagaInstance(saga, sagaMetadata, () => DateTime.UtcNow);
+
+            var currentSagaMetadata = sagaMetadataCollection.Find(context.MessageHandler.Instance.GetType());
+
+            string targetSagaTypeString;
+            string targetSagaId;
+
+            if (context.Headers.TryGetValue(Headers.SagaType, out targetSagaTypeString) && context.Headers.TryGetValue(Headers.SagaId, out targetSagaId))
```

Comment:
I remember some conversations about detecting when only a saga id header is present without the saga type header which should not be valid. Did you dismiss that approach?